docs/tutorials/embedding/MASE.ipynb

```diff
@@ -0,0 +1,374 @@
+    "$$\\hat{P}^{(i)} = V R^{(i)} V^T$$"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "R = embedder.scores_\n",
+    "R_1 = R[0]\n",
+    "R_2 = R[1]\n",
+    "\n",
+    "V_hat_T = np.transpose(V_hat)\n",
+    "P_1_hat = np.matmul(np.matmul(V_hat, R_1), V_hat_T)\n",
+    "P_2_hat = np.matmul(np.matmul(V_hat, R_2), V_hat_T)\n",
+    "\n",
+    "heatmap(P_1_hat, figsize=(7,7), title=\"Visualization of Probability Matrix 1\")\n",
+    "heatmap(P_2_hat, figsize=(7,7), title=\"Visualization of Probability Matrix 2\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Recover Original Block Probabilities"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, we can see the clear block structure once again.  In order to obtain an estmate of the original block probabilities, we can average across the blocks."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "P1_hat_block = np.array([[np.mean(P_1_hat[:25, :25]), np.mean(P_1_hat[:25, 25:])],\n",
+    "               [np.mean(P_1_hat[25:, :25]), np.mean(P_1_hat[25:, 25:])]])\n",
+    "P2_hat_block = np.array([[np.mean(P_2_hat[:25, :25]), np.mean(P_2_hat[:25, 25:])],\n",
+    "               [np.mean(P_2_hat[25:, :25]), np.mean(P_2_hat[25:, 25:])]])\n",
+    "print(\"P1_hat_block =\\n\", P1_hat_block)\n",
+    "print(\"P2_hat_block =\\n\", P2_hat_block)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    " Although we don't get exactly the same block probabilities as we input into the SBM, it's fairly close with only 1 graph for each!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Sampling More Graphs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can recover an even more accurate representation of the two SBMs with more sample graphs.  Let's try with 100 graphs sampled from each.  Interestingly, this time ``MultipleASE()`` chooses to embed into 4 dimensions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "G1 = [sbm(n, P1) for i in range(100)]\n",
+    "G2 = [sbm(n, P2) for i in range(100)]\n",
+    "\n",
+    "V_hat = embedder.fit_transform(G1 + G2)\n",
+    "print(V_hat.shape)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Visualizing Common Latent Subspace"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As before, the latent subspace shows two distinct clusters, especially in dimensions 1 vs dimension 2."
```

Comment:
dimension 3 and on are noise dimensions anyway, so there shouldn't be clusters in them. so it's not "especially", but only in those dimensions.